src/swap/components/SwapProvider.tsx

```diff
@@ -228,54 +239,42 @@ export function SwapProvider({
           response.to.decimals,
         );
         destination.setAmount(formattedAmount);
-        setLifeCycleStatus({
+        updateLifeCycleStatus({
           statusName: 'amountChange',
           statusData: {
             amountFrom: type === 'from' ? amount : formattedAmount,
             amountTo: type === 'to' ? amount : formattedAmount,
+            tokenFrom: from.token,
+            tokenTo: to.token,
             // if quote was fetched successfully, we
             // have all required fields
             isMissingRequiredField: !formattedAmount,
-            maxSlippage,
-            tokenFrom: from.token,
-            tokenTo: to.token,
           },
         });
       } catch (err) {
-        setLifeCycleStatus({
+        updateLifeCycleStatus({
           statusName: 'error',
           statusData: {
             code: 'TmSPc01', // Transaction module SwapProvider component 01 error
             error: JSON.stringify(err),
             message: '',
-            // LifecycleStatus shared data
-            isMissingRequiredField:
-              lifeCycleStatus.statusData.isMissingRequiredField,
-            maxSlippage,
           },
         });
       } finally {
         // reset loading state when quote request resolves
         destination.setLoading(false);
       }
     },
-    [from, lifeCycleStatus, to, useAggregator],
+    [from, to, lifeCycleStatus, updateLifeCycleStatus, useAggregator],
   );
 
   const handleSubmit = useCallback(async () => {
     if (!address || !from.token || !to.token || !from.amount) {
       return;
     }
-    const maxSlippage = lifeCycleStatus.statusData.maxSlippage;
-    setLifeCycleStatus({
-      statusName: 'init',
-      statusData: {
-        isMissingRequiredField: false,
-        maxSlippage,
-      },
-    });
```

Comment:
I don't believe this is needed, unless it is possible to swap even if the quote fails.  Status init doesn't make sense at this point when you are in the middle of the process.

Reply:
Yeah makes sense.